File: QTD2.Infrastructure/Model/Task_Review/TaskReviewActionItem_VM.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QTD2.Infrastructure.Model.Task_Review
{
    public class TaskReviewActionItem_VM
    {
        public int Id { get; set; }
        public string Type { get; set; }
        public int? AssigneeId { get; set; }
        public int PriorityId { get; set; }
        public DateTime AssignedDate { get; set; }
        public DateTime DueDate { get; set; }
        public string Notes { get; set; }
        public List<TaskReviewActionItem_SubDutyArea_Operation_VM> SubDuty_Operations { get; set; } = new List<TaskReviewActionItem_SubDutyArea_Operation_VM>();
        public int? Task_number { get; set; }
        public string? Task_statement { get; set; }
        public List<TaskReviewActionItem_Step_Operation_VM> Step_Operations { get; set; } = new List<TaskReviewActionItem_Step_Operation_VM>();
        public List<TaskReviewActionItem_QuestionAndAnswer_Operation_VM> QuestionAndAnswer_Operations { get; set; } = new List<TaskReviewActionItem_QuestionAndAnswer_Operation_VM>();
        public List<TaskReviewActionItem_Suggestion_Operation_VM> Suggestion_Operations { get; set; } = new List<TaskReviewActionItem_Suggestion_Operation_VM>();
        public string? Conditions { get; set; }
        public string? Criteria { get; set; }
        public string? References { get; set; }
        public List<TaskReviewActionItem_Tool_Operation_VM> Tool_Operations { get; set; } = new List<TaskReviewActionItem_Tool_Operation_VM>();
        public List<TaskReviewActionItem_EnablingObjective_Operation_VM> EnablingObjective_Operations { get; set; } = new List<TaskReviewActionItem_EnablingObjective_Operation_VM>();
        public List<TaskReviewActionItem_Procedure_Operation_VM> Procedure_Operations { get; set; }  = new List<TaskReviewActionItem_Procedure_Operation_VM>();
        public List<TaskReviewActionItem_RegulatoryRequirement_Operation_VM> RegulatoryRequirement_Operations { get; set; } = new List<TaskReviewActionItem_RegulatoryRequirement_Operation_VM>();
        public List<TaskReviewActionItem_SafetyHazard_Operation_VM> SafetyHazard_Operations { get; set; } = new List<TaskReviewActionItem_SafetyHazard_Operation_VM>();
        public bool? IsMeta { get; set; }

        public string? Priority { get; set; }
        public string? Assignees { get; set; }
    }
}

File: QTD.SPA/src/app/_DtoModels/Task_Review/TaskReviewActionItem_VM.ts
import { TaskReviewActionItem_EnablingObjective_Operation_VM } from "./TaskReviewActionItem_EnablingObjective_Operation_VM";
import { TaskReviewActionItem_Procedure_Operation_VM } from "./TaskReviewActionItem_Procedure_Operation_VM";
import { TaskReviewActionItem_QuestionAndAnswer_Operation_VM } from "./TaskReviewActionItem_QuestionAndAnswer_Operation_VM";
import { TaskReviewActionItem_RegulatoryRequirement_Operation_VM } from "./TaskReviewActionItem_RegulatoryRequirement_Operation_VM";
import { TaskReviewActionItem_SafetyHazard_Operation_VM } from "./TaskReviewActionItem_SafetyHazard_Operation_VM";
import { TaskReviewActionItem_Step_Operation_VM } from "./TaskReviewActionItem_Step_Operation_VM";
import { TaskReviewActionItem_SubDutyArea_Operation_VM } from "./TaskReviewActionItem_SubDutyArea_Operation_VM";
import { TaskReviewActionItem_Suggestion_Operation_VM } from "./TaskReviewActionItem_Suggestion_Operation_VM";
import { TaskReviewActionItem_Tool_Operation_VM } from "./TaskReviewActionItem_Tool_Operation_VM";

export class TaskReviewActionItem_VM{
         id:string;
         type:string;
         assigneeId:string;
         priorityId:string;
         assignedDate: Date;
         notes:string;
         dueDate:Date;
         subDuty_Operations:TaskReviewActionItem_SubDutyArea_Operation_VM[] = [];
         task_number:string;
         task_statement:string;
         step_Operations:TaskReviewActionItem_Step_Operation_VM[] = [];
         questionAndAnswer_Operations:TaskReviewActionItem_QuestionAndAnswer_Operation_VM[] = [];
         suggestion_Operations:TaskReviewActionItem_Suggestion_Operation_VM[] = [];
         conditions:string;
         criteria:string;
         references:string;
         tool_Operations:TaskReviewActionItem_Tool_Operation_VM[] = [];
         enablingObjective_Operations:TaskReviewActionItem_EnablingObjective_Operation_VM[] = [];
         procedure_Operations:TaskReviewActionItem_Procedure_Operation_VM[] = [];
         regulatoryRequirement_Operations:TaskReviewActionItem_RegulatoryRequirement_Operation_VM[] = [];
         safetyHazard_Operations:TaskReviewActionItem_SafetyHazard_Operation_VM[] = [];
         isMeta:boolean;
         priority:string;
         assignees:string;
}